File: freecodecamp/daily-code-challenge/2026-01-29.py
def separate_letters_and_numbers(s):
    if not s:
        return s

    result = [s[0]]

    for i in range(1, len(s)):
        prev = s[i - 1]
        curr = s[i]

        if (prev.isalpha() and curr.isdigit()) or (prev.isdigit() and curr.isalpha()):
            result.append('-')

        result.append(curr)

    return ''.join(result)

File: freecodecamp/daily-code-challenge/2026-01-29.js
function separateLettersAndNumbers(str) {
  if (!str) return str;

  let result = str[0];

  for (let i = 1; i < str.length; i++) {
    const prev = str[i - 1];
    const curr = str[i];

    const isPrevLetter = /[a-zA-Z]/.test(prev);
    const isCurrLetter = /[a-zA-Z]/.test(curr);
    const isPrevDigit = /[0-9]/.test(prev);
    const isCurrDigit = /[0-9]/.test(curr);

    if (
      (isPrevLetter && isCurrDigit) ||
      (isPrevDigit && isCurrLetter)
    ) {
      result += '-';
    }

    result += curr;
  }

  return result;
}
